indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -104,6 +105,10 @@ public final static InputRow parseInputRow(Object value, InputRowParser parser)
     if(parser instanceof StringInputRowParser && value instanceof Text) {
       //Note: This is to ensure backward compatibility with 0.7.0 and before
       return ((StringInputRowParser) parser).parse(value.toString());
+    } else if(parser instanceof StringInputRowParser && value instanceof BytesWritable) {
```

Comment:
Or may be write a core HadoopStringishInputRowParser (that extends StringInputRowParser) that takes Object type record as input and handles all the hadoopy stringish types possible in a big if-else. That way all the special handling will be that special place only and we can keep things a bit cleaner here.